src/swap/SwapScreen.tsx

```diff
@@ -300,14 +344,34 @@ export function SwapScreen({ route }: Props) {
       return
     }
 
-    if (parsedSwapAmount[Field.FROM].gt(maxFromAmount)) {
-      setFromSwapAmountError(true)
-      showMaxCeloSwapWarning()
-      dispatch(showError(t('swapScreen.insufficientFunds', { token: fromToken?.symbol })))
-    } else {
-      dispatch(setSwapUserInput(userInput))
-      navigate(Screens.SwapReviewScreen)
-    }
+    // Legacy
+
+    const swapResponse = exchangeRate.rawSwapResponse
+
+    ValoraAnalytics.track(SwapEvents.swap_screen_confirm_swap, {
+      toToken: toToken.address,
+      fromToken: fromToken.address,
+      amount: swapAmount[updatedField],
+      amountType: swapAmountParam,
+      usdTotal: new BigNumber(swapResponse.unvalidatedSwapTransaction[swapAmountParam])
+        .shiftedBy(-swapAmountDecimals)
+        .multipliedBy(swapResponse.unvalidatedSwapTransaction.price)
+        .toNumber(),
+      allowanceTarget,
+      estimatedPriceImpact,
+      price,
+      provider: swapResponse.details.swapProvider,
+      web3Library: 'contract-kit',
+    })
+
+    dispatch(setSwapUserInput(userInput))
+    dispatch(
```

Comment:
ahh, the dreaded double dispatch. i think both `swapStart*` actions already set the swapInfo, so we can just remove this action altogether?

Reply:
great catch, thanks! looks like `userInput` passed here is not used anywhere, and can be removed.

although, it seems logical to me to set `swapState` to `USER_INPUT` just after the swap screen is open, so I renamed and moved this reducer to the `useEffect` fired at the first render

since the code changes are spread all over the codebase, I created a separate PR https://github.com/valora-inc/wallet/pull/4509 to merge into this one